astra/src/main/java/com/slack/astra/chunk/ReadOnlyChunkImpl.java

```diff
@@ -128,6 +173,142 @@ public ReadOnlyChunkImpl(
     LOG.debug("Created a new read only chunk - zkSlotId: {}", slotId);
   }
 
+  /*
+  ======================================================
+  All methods below RELATED to astra.ng.dynamicChunkSizes
+  ======================================================
+   */
+
+  public void evictChunk(CacheNodeAssignment cacheNodeAssignment) {
```

Comment:
Try to limit visibility of these methods as much as possible (ie, use `private`, `protected` can help in writing tests). If it's only public to facilitate tests, consider adding the `@VisibleForTesting` annotation.